cogito/utils_test.go

```diff
@@ -0,0 +1,47 @@
+// This file contains test utilities.
+
+package cogito_test
+
+import (
+	"encoding/json"
+	"errors"
+	"testing"
+
+	"github.com/imdario/mergo"
+	"gotest.tools/v3/assert"
+)
+
+// toJSON returns the JSON encoding of thing.
+func toJSON(t *testing.T, thing any) []byte {
+	t.Helper()
+	buf, err := json.Marshal(thing)
+	assert.NilError(t, err)
+	return buf
+}
+
+// fromJSON unmarshals the JSON-encoded data into thing.
+func fromJSON(t *testing.T, data []byte, thing any) {
+	t.Helper()
```

Comment:
Update: I checked here: https://about.sourcegraph.com/blog/go/advanced-testing-in-go
and I assume answer is in this list:
```
Never return errors. Just pass in *testing.T and fail.
By not returning errors, usage is much prettier, since you don't have a bunch of visual overhead from error handling code in your tests since error checking is gone.
Used to make tests clear on what they’re testing vs what is boilerplate
Call t.Helper() for cleaner failure output (Go 1.9)
```